OlinkAnalyze/R/bridgeable.R

```diff
@@ -0,0 +1,293 @@
+      dplyr::group_by(.data[["Gene"]], .data[["ids"]]) |>
+      filter(.data[["Count_3k"]] > 10, .data[["Count_ht"]] > 10) |>
+      dplyr::summarise(ks = stats::ks.test(
+        .data[["NPX_ht"]], .data[["NPX_3k"]])$statistic, .groups = "keep") |>
+      dplyr::ungroup() |>
+      dplyr::distinct()
+
+    return(data_out)
+  }
+
+
+  e3k_eHT_mapping <- OlinkAnalyze:::eHT_e3072_mapping
+
+  data_Explore384 <- data_Explore384 |>
+    dplyr::filter(
+      stringr::str_detect(string = .data[["SampleType"]], pattern = "SAMPLE"),
+      stringr::str_detect(string = .data[["AssayType"]], pattern = "assay")
+    ) |>
+    map_oid_ht_3k() |>
+    dplyr::mutate(OlinkID_Explore384 = substr(.data[["OlinkID"]], 10, 18)) |>
+    dplyr::rename(
+      Gene = .data[["Assay"]],
+      NPX_3k = .data[["NPX"]],
+      ids = .data[["OlinkID"]]
+    ) |>
+    dplyr::select(.data[["SampleID"]],
+                  .data[["ids"]],
+                  .data[["OlinkID_Explore384"]],
+                  .data[["NPX_3k"]],
+                  .data[["Count"]]) |>
+    dplyr::rename(Count_3k = .data[["Count"]]) |>
+    distinct()
+
+  # Count refers to HT counts
+  data_HT <- data_HT |>
+    dplyr::filter(
+      stringr::str_detect(string = .data[["SampleType"]], pattern = "SAMPLE"),
+      stringr::str_detect(string = .data[["AssayType"]], pattern = "assay")
+    ) |>
+    map_oid_ht_3k() |>
+    dplyr::rename(
+      Gene = .data[["Assay"]],
+      NPX_ht = .data[["NPX"]],
+      ids = .data[["OlinkID"]]
+    ) |>
+    dplyr::mutate(OlinkID = substr(.data[["ids"]], 1, 8)) |>
+    dplyr::select(.data[["SampleID"]],
+                  .data[["Gene"]],
+                  .data[["OlinkID"]],
+                  .data[["ids"]],
+                  .data[["UniProt"]],
+                  .data[["NPX_ht"]],
+                  .data[["Count"]]) |>
+    dplyr::mutate(Count_ht = .data[["Count"]]) |>
+    distinct()
+
+  data_mapped <- data_Explore384 |>
+    dplyr::inner_join(data_HT, by = c("SampleID", "ids"),
+                      relationship = "many-to-many") |>
+    na.omit() |>
+    dplyr::group_by(.data[["ids"]]) |>
+    dplyr::mutate(range_ht =
+                    quantile(.data[["NPX_ht"]][.data[["Count_ht"]] > 10],
+                             probs = 0.9) -
+                    quantile(.data[["NPX_ht"]][.data[["Count_ht"]] > 10],
+                             probs = 0.1),
+                  # using 10% to 90% quantiles for checking range /IQR
+                  range_3k = quantile(.data[["NPX_3k"]][.data[["Count_3k"]] >
+                                                          10],
+                                      probs = 0.9) -
+                    quantile(.data[["NPX_3k"]][.data[["Count_3k"]] > 10],
+                             probs = 0.1),
+                  range_diff = abs(.data[["range_ht"]] - .data[["range_3k"]]),
+                  iqr_ht = IQR(.data[["NPX_ht"]]),
+                  iqr_3k = IQR(.data[["NPX_3k"]]),
```

Comment:
IQR is also computed in the function "outlier_removal_iqr". This is duplicate code.

Reply:
These are not used anywhere, removed